drivers/hmis/app/jobs/hmis/activity_log_processor_job.rb

```diff
@@ -0,0 +1,153 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# = Hmis ActivityLogProcessorJob
+#
+# Link the access logs for graphql object/fields to database ids for root entity types (client, enrollment, project).
+# This linking is done async to avoid overhead of doing it inside a request
+#
+module Hmis
+  class ActivityLogProcessorJob < ::BaseJob
+    def perform(force: false)
+      Hmis::ActivityLog.with_advisory_lock(
+        'AccessLogProcessorLock',
+        timeout_seconds: 0, # don't wait
+      ) do
+        scope = Hmis::ActivityLog.all
+        scope = scope.unprocessed unless force
+        process_records(scope)
+      end
+    end
+
+    protected
+
+    RESOLVE_ENROLLMENT_IDS = ->(enrollment_ids) {
+      e_t = Hmis::Hud::Enrollment.arel_table
+      c_t = Hmis::Hud::Client.arel_table
+      client_ids_map = Hmis::Hud::Enrollment.with_deleted.
+        where(id: enrollment_ids).
+        joins(
+          e_t.create_join(
+            c_t,
+            e_t.create_on(
+              # joins to clients; intentionally includes deleted records
+              [e_t[:PersonalID].eq(c_t[:PersonalID]), e_t[:data_source_id].eq(c_t[:data_source_id])].inject(&:and),
+            ),
+          ),
+        ).
+        pluck(e_t[:id], c_t[:id]).to_h
```

Comment:
Why all this complexity instead of `Hmis::Hud::Enrollment.with_deleted.joins(:client).merge(Hmis::Hud::Client.with_deleted).pluck(e_t[:id], c_t[:id])`?  On the warehouse side we also have built out `GrdaWarehouse::Hud::WithDeleted::Enrollment` etc. models, might be worth exploring here.

Reply:
That's fair. I'll try using merge() as you suggest. I'm not sure about adding models or associations just support access to deleted rows.

It's worth mentioning that I did figure out a way to globally disable the paranoia gem via an env variable. I reverted it since it felt kind of heavy just for this use-case.

Reply:
I'll take what you have over globally disabling paranoia, but I'm pretty sure the merge will work.

Reply:
Looks like merging with_deleted doesn't work as we might expect. Note the below sql is still filtering clients by DateDeleted which isn't what we want. (I think I'm invoking it correctly but I might be missing something)
```ruby
puts  Hmis::Hud::Enrollment.with_deleted.joins(:client).merge(Hmis::Hud::Client.with_deleted).select(e_t[:id]).to_sql
# => SELECT "Enrollment"."id" FROM "Enrollment" INNER JOIN "Client" ON "Client"."DateDeleted" IS NULL AND "Client"."data_source_id" = "Enrollment"."data_source_id" AND "Client"."PersonalID" = "Enrollment"."PersonalID"
```

Interestingly, it looks like the merged condition is applied to enrollment instead of client, so the following statements result in equivalent sql
```ruby
Hmis::Hud::Enrollment.joins(:client).merge(Hmis::Hud::Client.with_deleted).select(e_t[:id])
Hmis::Hud::Enrollment.with_deleted.joins(:client).select(e_t[:id])
# => true
```

I suspect this is because the paranoia gem is playing games with default_scope under the hood which seems gnarly.

Reply:
This seems to work:
```
Hmis::Hud::Client.unscoped { Hmis::Hud::Enrollment.joins(:client).merge(Hmis::Hud::Client.with_deleted).to_sql }
# SELECT "Enrollment".* FROM "Enrollment" INNER JOIN "Client" ON "Client"."data_source_id" = "Enrollment"."data_source_id" AND "Client"."PersonalID" = "Enrollment"."PersonalID"
```

It's not the most obvious solution.  Looks like:
```
Hmis::Hud::Client.unscoped { Hmis::Hud::Enrollment.with_deleted.joins(:client).to_sql }
# SELECT "Enrollment".* FROM "Enrollment" INNER JOIN "Client" ON "Client"."data_source_id" = "Enrollment"."data_source_id" AND "Client"."PersonalID" = "Enrollment"."PersonalID"
```
also works, so maybe that's a better/simpler implementation.  Basically you can pass `unscoped` a block for a given model and it'll remove the paranoia restriction.

Reply:
I don't like this as much, but it looks like you can nest the unscoped blocks.
```
Hmis::Hud::Client.unscoped { Hmis::Hud::Enrollment.unscoped { puts Hmis::Hud::Enrollment.joins(:client).to_sql }}
# SELECT "Enrollment".* FROM "Enrollment" INNER JOIN "Client" ON "Client"."data_source_id" = "Enrollment"."data_source_id" AND "Client"."PersonalID" = "Enrollment"."PersonalID"
```

Reply:
I didn't realize unscoped takes a block, good to know! I'd say that seems reasonable as long as we are confident there's never going to be anything other than paranoia managing the default scope of these models; then we can treat Model.unscoped as equivalent to with_deleted. I'll make that change